File: fluentcorej9/src/main/java/com/mcreater/fxfluent/util/DefaultFontPatcher.java
package com.mcreater.fxfluent.util;

import javafx.scene.text.Font;

import java.lang.reflect.Field;

public class DefaultFontPatcher {
    public static void patch(Font font) throws NoSuchFieldException, IllegalAccessException {
        System.setProperty("prism.lcdtext", "false");
        if (!ReflectUtil.openModuleToAllUnnamed(
                Font.class.getModule(),
                "javafx.scene.text"
        )) {
            System.out.println("Patch failed");
            return;
        }

        Field field = Font.class.getDeclaredField("DEFAULT");
        field.setAccessible(true);
        field.set(null, font);
    }
}

File: fluentcorej9/src/main/java/com/mcreater/fxfluent/util/DefaultFontPatcher.kt
package com.mcreater.fxfluent.util

import com.mcreater.fxfluent.util.ReflectUtil.Companion.openModuleToAllUnnamed
import javafx.scene.text.Font

class DefaultFontPatcher {
    companion object {
        @JvmStatic
        @Throws(NoSuchFieldException::class, IllegalAccessException::class)
        fun patch(font: Font?) {
            if (!openModuleToAllUnnamed(
                    Font::class.java.module,
                    "javafx.scene.text"
                )
            ) {
                println("Patch failed")
                return
            }
            val field = Font::class.java.getDeclaredField("DEFAULT")
            field.setAccessible(true)
            field[null] = font
        }
    }
}